charts/presto/templates/configmap-worker.yaml

```diff
@@ -0,0 +1,23 @@
+{{- if not (eq .Values.mode "single") }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .Release.Name }}-worker
+  labels:
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    app.kubernetes.io/name: {{ .Chart.Name }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/version: {{ .Values.image.tag | default .Chart.AppVersion | quote }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+data:
+  node.properties: |-
+    node.environment={{ .Values.node.environment }}
+    node.data-dir={{ .Values.node.dataDir }}
+  jvm.config: |-
+    {{- .Values.worker.jvm | default .Values.jvm | nindent 4 }}
+  config.properties: |-
+    coordinator=false
+    http-server.http.port={{ .Values.service.port }}
+    discovery.uri=http://{{ .Release.Name }}:{{ .Values.service.port }}
+    {{- .Values.config | nindent 4 }}
```

Comment:
Add the `presto.version=`, as if this is not defined/hardcoded in the config can sometimes produce hard to debug connection issues:
```yaml
presto.version={{ .Chart.AppVersion }}
```

Reply:
In general `presto.version != {{ .Chart.AppVersion }}`, because Presto distributive inside docker image represents the actual version. I mean I can install Presto `0.276` version while `{{ .Chart.AppVersion }} == 0.278`.

BTW there are two `SELECT * FROM system.runtime.nodes` results for the chart installations show correct `node_version`:
```
helm install my-presto charts/presto

+------------------------------------+----------------------+-------------+-----------+------+
|node_id                             |http_uri              |node_version |coordinator|state |
+------------------------------------+----------------------+-------------+-----------+------+
|82c21439-ecd7-46c2-9a1d-3714f44abe5a|http://172.17.0.6:8080|0.278-6156edd|false      |active|
|d0a8906c-1665-483f-92b3-f6ab0c128028|http://172.17.0.7:8080|0.278-6156edd|false      |active|
|f0bb7f12-9aec-4ed6-82bd-7071b896e07e|http://172.17.0.8:8080|0.278-6156edd|true       |active|
+------------------------------------+----------------------+-------------+-----------+------+
```

```
helm install my-presto charts/presto --set image.tag=0.278.1

+------------------------------------+----------------------+---------------+-----------+------+
|node_id                             |http_uri              |node_version   |coordinator|state |
+------------------------------------+----------------------+---------------+-----------+------+
|309a45a0-8a8f-494c-a84e-8ae14a6785a5|http://172.17.0.7:8080|0.278.1-ec67ba1|false      |active|
|56e8a0f1-b848-4be7-b596-df40f8c97bc2|http://172.17.0.6:8080|0.278.1-ec67ba1|true       |active|
|e75be05e-9528-4a44-8765-ed9505036ed6|http://172.17.0.8:8080|0.278.1-ec67ba1|false      |active|
+------------------------------------+----------------------+---------------+-----------+------+
```

Reply:
No making this configurable nor setting this explicitly will produce issues for un-aware users making it hard to debug. Presto (not sure if this have changed or not) will not allow nodes with other version then the coordinator to the cluster. So in case one would like to connect Prestissimo to a JAVA Coordinator it will fail - that is why I would propose to set this explicitly based on chart version.

Reply:
It's a configurable property in the current implementation.
Probably it's ok to add it to default config, but It's not clear to me what is the value should be used by default.